<launch>
  <arg name="rgb_topic" default="/camera/rgb/image_raw" />
  <arg name="depth_registered_topic" default="/camera/depth/image_raw"/>
  <arg name="depth_camera_info_topic" default="/camera/depth/camera_info" />
  <arg name="rgb_camera_info_topic" default="/camera/rgb/camera_info" />

  <node name="mask_rcnn" pkg="mask_rcnn_ros" type="mask_rcnn_node.py" output="screen">
      <remap from="~input" to="$(arg rgb_topic)" />
      <param name="~visualization" value="true" />
  </node>

  <node name="depth_segmentation_node" pkg="depth_segmentation" type="depth_segmentation_node" output="screen">
    <param name="depth_image_sub_topic" value="$(arg depth_registered_topic)" />
    <param name="rgb_image_sub_topic" value="$(arg rgb_topic)" />
    <param name="depth_camera_info_sub_topic" value="$(arg depth_camera_info_topic)" />
    <param name="rgb_camera_info_sub_topic" value="$(arg rgb_camera_info_topic)" />
    <param name="semantic_instance_segmentation/enable" value="false"/>
  </node>
</launch>
